packages/tinacms/src/admin/pages/CollectionCreatePage.tsx

```diff
@@ -145,19 +143,19 @@ const RenderForm = ({ cms, collection, templateName, mutationInfo }) => {
 
   // the schema is being passed in from the frontend so we can use that
   const schemaCollection = schema.getCollection(collection.name)
-  const template: GlobalTemplate<true> = schema.getTemplateForData({
+  const template = schema.getTemplateForData({
     collection: schemaCollection,
     data: { _template: templateName },
-  }) as GlobalTemplate<true>
+  })
 
-  const formInfo = resolveForm({
+  const formInfo = schema.resolveForm({
     collection: schemaCollection,
     basename: schemaCollection.name,
     schema: schema,
-    template,
+    templateName: template.name,
   })
 
-  let slugFunction = template?.ui?.filename?.slugify
+  let slugFunction = schemaCollection?.ui?.filename?.slugify
```

Comment:
`template` should never have these properties on it, but `getTemplateForData` returns a template _or_ a collection depending on if the collection has templates or fields. This just makes it more clear that we're only getting this value off of the collection, if it's present.